pkg/storage/db.go

```diff
@@ -428,15 +428,24 @@ func (db *DB) StoreV2(ctx context.Context, cloudAssetChanges domain.CloudAssetCh
 					break
 				}
 			}
+			if err != nil {
+				break
+			}
 			for _, ip := range val.PublicIPAddresses {
 				for _, hostname := range val.Hostnames { //TODO look very closely into awsconfig-tranformerd logic for this
 					if strings.EqualFold(added, val.ChangeType) {
 						err = db.assignPublicIP(ctx, tx, arnID, ip, hostname, cloudAssetChanges.ChangeTime)
 					} else {
 						err = db.releasePublicIP(ctx, tx, arnID, ip, hostname, cloudAssetChanges.ChangeTime)
 					}
+					if err != nil {
+						break
+					}
 				}
 			}
+			if err != nil {
```

Comment:
nit: i dont think this is necessary.

Reply:
Could you clarify why?

Reply:
I was referring more to pattern of using `break` in case of error. There are multiple ways in which we can make this pattern better:

1. Extract for loops into functions and return error or empty values in error case
2. Use Goto keyword, not clean but better than current pattern
3. Use Label to exit out of inner and outer loops

Reply:
That's fair, it's not the prettiest looking code. I am skeptical about using goto and labels, those feel very similar to me and I believe are generally discouraged for readability's sake. Splitting out the for loop into a function is an option, I can try to do that!